lib/capistrano/tasks/assets.rake

```diff
@@ -88,10 +88,15 @@ namespace :deploy do
     task :restore_manifest do
       on release_roles(fetch(:assets_roles)) do
         within release_path do
-          target = detect_manifest_path
-          source = release_path.join('assets_manifest_backup', File.basename(target))
-          if test "[[ -f #{source} && -f #{target} ]]"
-            execute :cp, source, target
+          targets = detect_manifest_path.split(' ')
+          sources = targets.map do |target|
+            release_path.join('assets_manifest_backup', File.basename(target))
+          end
+          if test(:ls, sources) && test(:ls, targets)
```

Comment:
I am surprised this works to pass an array as the second argument to `test`. My guess that `test` is calling `flatten` at some point, but I am not sure if that is documented behavior. What do you think about expanding the array to make this more explicit?

```suggestion
          if test(:ls, *sources) && test(:ls, *targets)
```